cpp/cgi_progs/zotero_cgi.cc

```diff
@@ -416,16 +416,20 @@ bool ProcessShowQASubActionAdd(const std::multimap<std::string, std::string> &cg
     // sub-action "add"
     const std::string add_type(GetCGIParameterOrDefault(cgi_args, "add_type"));
     const std::string add_tag(GetCGIParameterOrDefault(cgi_args, "add_tag"));
+    const std::string add_subfield_code(GetCGIParameterOrDefault(cgi_args, "add_subfield_code"));
     const std::string add_presence(GetCGIParameterOrDefault(cgi_args, "add_presence"));
-    if (add_type.empty() or add_tag.empty() or add_presence.empty())
+    const std::string add_record_type(GetCGIParameterOrDefault(cgi_args, "add_record_type"));
+    if (add_type.empty() or add_tag.empty() or add_subfield_code.empty() or add_presence.empty() or add_record_type.empty())
         return false;
 
     std::string journal_id_to_insert = db_connection->escapeAndQuoteString(zeder_journal_id);
     if (add_type == "global")
         journal_id_to_insert = "NULL";
 
-    db_connection->queryOrDie("INSERT INTO metadata_presence_tracer (zeder_journal_id, metadata_field_name, field_presence) "
+    db_connection->queryOrDie("INSERT INTO metadata_presence_tracer (zeder_journal_id, metadata_field_name, subfield_code, field_presence, record_type) "
                               " VALUES (" + journal_id_to_insert + ", " + db_connection->escapeAndQuoteString(add_tag) + ", "
+                              db_connection->escapeAndQuoteString(add_subfield_code) + ", "
```

Comment:
if add_subfield_code is empty, we want to set the value to NULL and not to an empty string